### PR TITLE
Notebooks: Support User-Installed Kernels

### PR DESCRIPTION
**Pull request by chlafreniere**
Given the work that we've shipped in April relaxing requirements around forcing reinstall of python packages (in addition to leveraging sparkmagic from a more widely-used repo), this allows us to open up ADS to more kernels than just our standard kernels that we provide. In this PR, I'm not addressing the "how do we query kernels without an active sessionManager (and thus, a serverManager)" issue (which we definitely need to discuss), and I don't want users to see other kernels if they don't want to at the moment, until we've tested this more broadly. Because of this, I have this functionality behind a setting.

As a test case, I used the dotnet interactive kernel (https://github.com/dotnet/interactive). There are a couple of places where I'm mucking with what environment variables that we're passing to Jupyter; turns out, vscode likes to change this particular variable for liveshare scenarios, and this location doesn't have some ASP.NET dependencies that the dotnet interactive kernel needs. Clearing out this variable allows those kernels to start as expected.

### CODE REVIEW

File: src/sql/workbench/contrib/notebook/browser/notebook.contribution.ts [hunk 1]
```diff
@@ -197,6 +197,19 @@ configurationRegistry.registerConfiguration({
...
 
+configurationRegistry.registerConfiguration({
+	'id': 'notebook',
+	'title': 'Notebook',
+	'type': 'object',
+	'properties': {
+		'notebook.showAllKernels': {
+			'type': 'boolean',
+			'default': false,
+			'description': localize('notebook.showAllKernels', "(Experimental) show all kernels for the current notebook provider.")
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/browser/notebook.contribution.ts:208]: Should be preview instead to keep consistent with our other "early" features

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/notebook/browser/notebook.contribution.ts:208]: Also this should be under the preview flag - in which case is this option even really necessary? Not sure if there's a case for someone wanting preview features enabled but not wanting to see these.

Comment (chlafreniere) [reply src/sql/workbench/contrib/notebook/browser/notebook.contribution.ts:208]: I really do think that we shouldn't turn this on for everyone. For example, by default, users will see 2 Python kernels with Anaconda (including one that has a crazy string as the display name), and so I really think we should roll this out slowly and methodically, until we have more data.

Totally hear you on the Experimental vs Preview thing. I'll make that change.

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/notebook/browser/notebook.contribution.ts:208]: Alright. This still should be controlled by the preview features flag as well though

**Pull request merged**